contracts/Core/BlockManager.sol

```diff
@@ -50,17 +51,17 @@ contract BlockManager is ACL, BlockStorage {
         _;
     }
 
-    function init(
-        address _stakeManagerAddress,
-        address _stateManagerAddress,
-        address _voteManagerAddress,
-        address _jobManagerAddress
-    ) external 
+    function initialize (
+        address _stakeManager,
+        address _stateManager,
+        address _voteManager,
+        address _jobManager
+    ) external initializer onlyRole(DEFAULT_ADMIN_ROLE)
     {
-        stakeManager = IStakeManager(_stakeManagerAddress);
-        stateManager = IStateManager(_stateManagerAddress);
-        voteManager = IVoteManager(_voteManagerAddress);
-        jobManager = IJobManager(_jobManagerAddress);
+        stakeManager = IStakeManager(_stakeManager);
```

Comment:
why was _stakeManagerAddress renamed to _stakeManager
The "Address" part should be there. stakeManager and _stakeManager are different data types.

Reply:
same for other variables

Reply:
>  The "Address" part should be there. stakeManager and _stakeManager are different data types.

Since typecasting it to the same type is being done immediately, I don't think it's an issue. But there are other reasons why I renamed this - 

- According to solidity's default code-style convention, the underscore prepended with identifier only if there exists a storage variable with that name or a function/state variable is private. (This guide is highly subjective, I am referencing a default recommendation).

- Yes initially the type of `initialize` args are different but imminent type-casting is done.

Solution - Either rename the args to `stakeManagerAddress` or `_stakeManager`, I prefer latter.

Reply:
I would prefer stakeManagerAddress since it is more descriptive and less confusing. Please avoid changes unrelated to the issue in the future.